packages/store/addon/-private/system/store.ts

```diff
@@ -2745,27 +2663,29 @@ const Store = Service.extend({
     return relationships;
   },
 
-  _internalModelForResource(resource) {
-    let internalModel;
-    if (resource.clientId) {
-      internalModel = this._newlyCreatedModelsFor(resource.type).get(resource.clientId);
-    }
-    if (!internalModel) {
-      internalModel = this._internalModelForId(resource.type, resource.id);
-    }
-    return internalModel;
+  _internalModelForResource(resource: JsonApiResourceIdentity): InternalModel {
+    return internalModelFactoryFor(this).getByResource(resource);
+  },
+
+  /**
+   * TODO Only needed temporarily for test support
+   *
+   * @internal
+   */
+  _internalModelForId(modelName: string, id: string | null, lid: string | null): InternalModel {
+    return internalModelFactoryFor(this).lookup(modelName, id, lid);
```

Comment:
If this is still the old `clientId` concept it might be misleading to call it `lid`